### PR TITLE
Submit Week1 homework - Oguzhan

### PR DESCRIPTION
**Pull request by oguzhanturgut**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -23,24 +23,147 @@
       const value = options[key];
       if (key === 'text') {
         elem.textContent = value;
+      } else if (key === 'html') {
+        elem.innerHTML = value;
```
Comment (remarcmij) [inline homework/index.js]: By adding an option to set the `innerHTML` this function has now become ambiguous in the sense that it is no longer clear which of the two element-creation methods is to be used: `document.createElement()` or `innerHTML`.

File: homework/index.js [hunk 2]
```diff
@@ -23,24 +23,147 @@
...
     });
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  const renderError = error => {
```
Comment (remarcmij) [inline homework/index.js]: Here you have chosen to use an (arrow) function expression assigned to a `const`. In other places there are regular function declarations, with the `function` keyword. It is best to be consistent and choose one style or the other.

My personal preference is to use regular function declarations for (named) functions at the file/global level and use (anonymous) arrow functions for callbacks and event handlers. This makes it easy to discriminate between data constants (defined with `const`) and functions (declared with `function`) at the file/global level.

File: homework/index.js [hunk 3]
```diff
@@ -23,24 +23,147 @@
...
+    const wrapper = document.getElementById('wrapper');
+    wrapper.innerHTML = '';
+    createAndAppend('div', wrapper, {
+      text: error.message,
+      class: 'alert',
+    });
+  };
+
+  const addRepoInfoRow = (parent, label, content) => {
```
Comment (remarcmij) [inline homework/index.js]: 👍 for doing this in a separate function.

File: homework/index.js [hunk 4]
```diff
@@ -23,24 +23,147 @@
...
+    const row = createAndAppend('tr', parent);
+    createAndAppend('th', row, {
+      text: label,
+      class: 'table-header',
+    });
+    createAndAppend('td', row, {
+      html: content,
```
Comment (remarcmij) [inline homework/index.js]: Can you make this work without reverting to using `innerHTML` in `createAndAppend`? You only need it for the `<a>` tag.

File: homework/index.js [hunk 5]
```diff
@@ -23,24 +23,147 @@
...
+
+  function renderRepoInfo(repository) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const repoTable = createAndAppend('table', leftSide);
+    const repoTableBody = createAndAppend('tbody', repoTable);
+    addRepoInfoRow(
+      repoTableBody,
+      'Repository',
+      `<a href="${repository.html_url}" target="_blank">${repository.name}</a>`,
```
Comment (remarcmij) [inline homework/index.js]: You could set the `text` attribute to the empty string `''` and 'fix up' the last child if the table row after your call to `addRepoInfoRow`:

```js
const row = addRepoInfoRow(repoTableBody, 'Repository', '');
createAndAppend('a', row.lastChild, {
  href: repository.html_url,
  target: '_blank',
  text: repository.name,
});
```

Of course, your `addRepoInfoRow` function should then return the row it created as its return value.

File: homework/index.js [hunk 6]
```diff
@@ -23,24 +23,147 @@
...
+        return;
+      }
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      repositories.forEach((repository, index) => {
+        createAndAppend('option', select, {
+          value: index,
+          text: repository.name,
+        });
+      });
+      createAndAppend('main', wrapper, {
```
Comment (remarcmij) [inline homework/index.js]: Semantically speaking, you should only have one `main` element in your HTML page. I think `section` would be better here.

File: homework/index.js [hunk 7]
```diff
@@ -23,24 +23,147 @@
...
+        class: 'left-side',
+        id: 'left-side',
+      });
+      renderRepoInfo(repositories[select.value]);
+      createAndAppend('main', wrapper, {
```
Comment (remarcmij) [inline homework/index.js]: Same here.

File: homework/style.css [hunk 8]
```diff
@@ -1,3 +1,127 @@
-.alert-error {
```
Comment (remarcmij) [inline homework/style.css]: On a mobile phone the boxes appear to be clipped on the right-hand side.

image

File: homework/index.js (hunk 1 above)
Comment (oguzhanturgut) [reply homework/index.js]: I reverted the createAndAppend function to the unmodified version. You are right that my version was going outside of its concept somehow.

File: homework/index.js (hunk 2 above)
Comment (oguzhanturgut) [reply homework/index.js]: I changed all to regular function declaration.

File: homework/index.js (hunk 3 above)
Comment (oguzhanturgut) [reply homework/index.js]: Thank you 🙏

File: homework/index.js (hunk 4 above)
Comment (oguzhanturgut) [reply homework/index.js]: I reverted `createAndAppend` function to its original state.

File: homework/index.js (hunk 5 above)
Comment (oguzhanturgut) [reply homework/index.js]: That is very useful way for escaping from using `html` in `createAndAppend` function. Many thanks for the tip.

File: homework/index.js (hunk 6 above)
Comment (oguzhanturgut) [reply homework/index.js]: Changed to section tag. Thanks for your advice

File: homework/style.css (hunk 8 above)
Comment (oguzhanturgut) [reply homework/style.css]: Fixed the styling problem on mobile screens. It now shows as expected

**Comment by oguzhanturgut** [comment]
Hello @remarcmij thanks for making time to review the PR. See you on Sunday 👋
Best Regards

**Pull request closed without merging**